pkg/dockerregistry/server/storagedriver.go

```diff
@@ -0,0 +1,32 @@
+package server
+
+import (
+	"regexp"
+
+	context "github.com/docker/distribution/context"
+	storagedriver "github.com/docker/distribution/registry/storage/driver"
+	registrystorage "github.com/docker/distribution/registry/storage/driver/middleware"
+)
+
+func init() {
+	registrystorage.Register("openshift", registrystorage.InitFunc(newLocalStorageDriver))
+}
+
+type localStorageDriver struct {
+	storagedriver.StorageDriver
+}
+
+var _ storagedriver.StorageDriver = &localStorageDriver{}
+
+var pathSplitRe = regexp.MustCompile("^(?P<repoPrefix>.*/repositories)/(?P<name>[^/]+/[^/]+)/(?P<content>.*)$")
```

Comment:
The exposition of `pathFor` would be great but it's only possible with a carry patch. I don't find it uglier than substituting blob path using regexps though.